drivers/eeprom/eeprom_fm25w256.c

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (c) 2021 Caspar Friedrich <c.s.w.friedrich@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <devicetree.h>
+#include <devicetree/gpio.h>
+#include <drivers/eeprom.h>
+#include <drivers/gpio.h>
+#include <drivers/spi.h>
+#include <logging/log.h>
+
+#define DT_DRV_COMPAT cypress_fm25w256
+
+// 256-Kbit ferroelectric random access memory logically organized as 32K × 8 bit
+#define SIZE KB(32)
+#define WORD_SIZE 8
+
+LOG_MODULE_REGISTER(fm25w256, CONFIG_EEPROM_LOG_LEVEL);
+
+// Opcode Commands
+enum {
+	OPCODE_WREN = 0b00000110,
+	OPCODE_WRDI = 0b00000100,
+	OPCODE_RDSR = 0b00000101,
+	OPCODE_WRSR = 0b00000001,
+	OPCODE_READ = 0b00000011,
+	OPCODE_WRITE = 0b00000010,
+};
+
+struct config {
+	struct spi_dt_spec spi_spec;
+	struct gpio_dt_spec wp_spec;
+	struct gpio_dt_spec hold_spec;
+	bool read_only;
+};
+
+static int configure_pin(const struct gpio_dt_spec *spec)
+{
+	if (!spec->port) {
+		return -ENODEV;
+	}
+
+	if (!device_is_ready(spec->port)) {
+		return -EACCES;
```

Comment:
This should be `-ENODEV` or maybe `-EIO`; this is incoherent across drivers. most return `-ENODEV` here; at24 drivers need to have this fixed, also.